src/dolphin/downloadDolphin.ts

```diff
@@ -10,15 +10,27 @@ import path from "path";
 import { lt } from "semver";
 
 import { fileExists } from "../main/fileExists";
-import { getLatestRelease } from "../main/github";
+import { fetchLatestDolphin } from "./checkVersion";
 import { ipc_dolphinDownloadFinishedEvent, ipc_dolphinDownloadLogReceivedEvent } from "./ipc";
-import { DolphinLaunchType } from "./types";
+import { DolphinLaunchType, DolphinVersionResponse } from "./types";
 import { findDolphinExecutable } from "./util";
 
 function logDownloadInfo(message: string): void {
   void ipc_dolphinDownloadLogReceivedEvent.main!.trigger({ message });
 }
 
+export async function assertDolphinInstallations(): Promise<void> {
+  try {
+    await assertDolphinInstallation(DolphinLaunchType.NETPLAY, logDownloadInfo);
+    await assertDolphinInstallation(DolphinLaunchType.PLAYBACK, logDownloadInfo);
```

Comment:
Ideally we could use Promise.all so we could potentially avoid a waterfall. However, since we're sending the log to `logDownloadInfo` we might not be able to do that, since it would intersperse the log messages. Something we could keep in mind for the future though.